cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo_auth.go

```diff
@@ -533,16 +587,32 @@ func newLocalOpaqueSecret(ownerRepo types.NamespacedName) *apiv1.Secret {
 	}
 }
 
-// "Local" in the sense of no namespace is specified
-func newLocalDockerConfigJsonSecret(ownerRepo types.NamespacedName) *apiv1.Secret {
-	return &apiv1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: ownerRepo.Name + "-",
-			Annotations: map[string]string{
-				annotationManagedByKey: annotationManagedByValue,
+func encodeDockerAuth(credentials *corev1.DockerCredentials) ([]byte, error) {
+	config := &credentialprovider.DockerConfigJSON{
+		Auths: map[string]credentialprovider.DockerConfigEntry{
+			credentials.Server: {
+				Username: credentials.Username,
+				Password: credentials.Password,
+				Email:    credentials.Email,
 			},
 		},
-		Type: apiv1.SecretTypeDockerConfigJson,
-		Data: map[string][]byte{},
 	}
+	return json.Marshal(config)
+}
+
+func decodeDockerAuth(dockerjson []byte) (*corev1.DockerCredentials, error) {
+	config := &credentialprovider.DockerConfigJSON{}
+	if err := json.Unmarshal(dockerjson, config); err != nil {
+		return nil, err
+	}
+	for server, entry := range config.Auths {
+		docker := &corev1.DockerCredentials{
+			Server:   server,
+			Username: entry.Username,
+			Password: entry.Password,
+			Email:    entry.Email,
+		}
+		return docker, nil
+	}
```

Comment:
I don't understand why you're using a for loop over the range of `config.Auths` here? It reads as though you're just returning after using the first of `config.Auths` always? But even then, there is no concept of "first" for a map in go (`config.Auths` is a `map[string]DockerConfigEntry` I think?) - the [order when you iterate a map in go is *explicitly* random](https://go.dev/doc/go1#iteration).

Reply:
I reused existing code. i.e. that is how Kubeapps has been handling AppRepository image pull secrets and docker config, assuming there was only one even though the struct is a collection.
is there a better way to take one item from a map than this kind of loop?